src/uthash.h

```diff
@@ -1098,6 +1191,9 @@ typedef struct UT_hash_handle {
    void *key;                        /* ptr to enclosing struct's key  */
    unsigned keylen;                  /* enclosing struct's key len     */
    unsigned hashv;                   /* result of hash-fcn(key)        */
+#if HASH_NOMEM_OK == 1
+   int mem_failed;
+#endif
```

Comment:
I definitely don't think the struct layout of `UT_hash_handle` should change based on `-DHASH_NOMEM_OK`; the name of the macro doesn't *sound* like it would affect the ABI.

Why does this flag need to live in the hash handle? Couldn't it live in `UT_hash_table` instead? This would still be an ABI breakage, but it wouldn't also be increasing the size of the user's structs; it would be `O(1)` extra space instead of `O(n)` extra space.  And actually `UT_hash_table` has plenty of extra padding bits on 64-bit platforms and/or when `HASH_BLOOM` is defined, some of which could be filled with a `uint8_t mem_failed` flag.